samples/lib/thrift/SSLsocket/hello_server/src/main.cpp

```diff
@@ -0,0 +1,88 @@
+/*
+ * Copyright 2022 Meta
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifdef __ZEPHYR__
+#include <zephyr.h>
+#endif
+
+#include <cstdio>
+#include <cstdlib>
+
+#include <thrift/protocol/TBinaryProtocol.h>
+#include <thrift/server/TSimpleServer.h>
+#include <thrift/transport/TBufferTransports.h>
+#include <thrift/transport/TSSLServerSocket.h>
+#include <thrift/transport/TSSLSocket.h>
+
+#include "Hello.h"
+#include "HelloHandler.h"
+
+using namespace ::apache::thrift;
+using namespace ::apache::thrift::protocol;
+using namespace ::apache::thrift::transport;
+using namespace ::apache::thrift::server;
+
+#ifdef __ZEPHYR__
+int main(void)
+#else
+int main(int argc, char** argv)
+#endif
+{
+  std::string my_addr;
+
+#ifdef __ZEPHYR__
+  my_addr = CONFIG_NET_CONFIG_MY_IPV4_ADDR;
+#else
+  if (argc < 2) {
+    printf("usage: %s <ip>\n", argv[0]);
+    return EXIT_FAILURE;
+  }
+
+  my_addr = std::string(argv[1]);
+#endif
+
+  int port = 4242;
+  std::shared_ptr<HelloHandler> handler(new HelloHandler());
+  std::shared_ptr<TBufferedTransportFactory> transportFactory(new TBufferedTransportFactory());
+  std::shared_ptr<TProtocolFactory> protocolFactory(new TBinaryProtocolFactory());
+  std::shared_ptr<TProcessor> processor(new HelloProcessor(handler));
+  std::shared_ptr<TSSLSocketFactory> socketFactory(new TSSLSocketFactory());
+  socketFactory->server(true);
+#ifdef __ZEPHYR__
```

Comment:
I think this logic is wrong - this file is always the server whether or it is being built for Zephyr or the host.